stellargraph/data/unsupervisedSampler.py

```diff
@@ -0,0 +1,229 @@
+
+        if number_of_walks < 1:
+            raise ValueError(
+                "({}) At least 1 walk from each head node has to be done".format(
+                    type(self).__name__
+                )
+            )
+        else:
+            self.number_of_walks = number_of_walks
+
+        self.seed = seed
+
+    def generator(self, batch_size):
+
+        """
+        This method yields a batch_size of +/- samples when it is called. The batch_size must be provided. It should be a positive integer. Moreover, for now we make the batch_size even by
+        unit incrementing it, incase it is odd. This is because we generate one negative sample for each positive sample.
+        This method:
+            1. Iterates over the root nodes.
+                1. Generates one walk at a time based on the walker method of the class  over the given Stellargraph. 
+                2. Incrementally extract positive (target,context) from the walks.
+                    1. For each positive pair sample a negative (target, context) from the graph based on a sampling distribution.
+                3. Once the total positive and negative samples generated are equal to the batch_size, the generator returns the samples. 
+         The positive samples are identical to GraphSAGE way of sampling. The negative sampling uses identical sampling distribution. 
+         However, no. of negative samples generated is perhaps different. This observation is based on our present understanding of Hamilton's implementation of GraphSAGE from github.
+         Note, we do not check if the negative sample is a positive context or a target itself. Although we do check that target is not its own context.
+         
+         Args:
+             batch_size: <int> number of samples to generate in each call of generator
+
+        Returns:
+            A tuple of lists: (target, context) pairs and their labels. 
+        """
+
+        self._check_parameter_values(batch_size=batch_size)
+
+        # making the batch_size even since we generate 1 negative sample for each positive one.
+        if batch_size % 2 != 0:
+            batch_size += 1
+
+        positive_pairs = list()
+        negative_pairs = list()
+
+        positive_samples_counter = 0
+        negative_samples_counter = 0
+
+        all_nodes = list(self.graph.nodes())
+
+        # Use the sampling distribution as per node2vec
+        degrees = self.graph.degree()
+        sampling_distribution = [degrees[n] ** 0.75 for n in all_nodes]
+
+        done = False
+        while not done:
+            for node in self.nodes:  # iterate over root nodes
+                # Get 1 walk at a time. For now its assumed that its a uniform random walker
+                walk = self.walker.run(
+                    nodes=[node],  # root nodes
+                    length=self.length,  # maximum length of a random walk
+                    n=1,  # number of random walks per root node
+                    seed=None,
```

Comment:
why don't you pass the seed from the constructor to the `walker.run` method?

Reply:
Now the self.seed value is being passed to the run method.